### DISCUSSION TITLE
export json from library

### DISCUSSION THREAD

**Discussion by markgoho** [nrwl/nx#5808] [General]
is it possible to export (and import) .json files from a library?

I know it's possible to import json in TypeScript, but I'm wondering specifically if there's a way to do this:
`import * as en from '@my-org/i18n'`

in the `lib/index.ts` file I'm trying to `export * from 'en.json'` but TypeScript does not like that at all.

here's my working solution:
``` typescript
import * as en from './lib/assets/i18n/en.json';
export { en };
```

And wherever I might need it, I'm essentially bringing it in as a JavaScript object, and if I needed it back to JSON I'd `stringify` it. What do you all think?

**Comment by SDemonUA** [comment]
Hi there, I'm trying to solve similar problem with translations. So as for now my approach is: 
 for front apps - copy jsons as assets on build stage via assets configuration
 for back apps - import as builded library, so it will not argue me to switch importJsonModules in tsconfig.

There only problem is how to assure my lib to be rebuild each time (specially when using nx serve)

**Reply by deefactorial** [reply]
I suggest for translations that instead of loading them at compile time in the build of your application that you load them async through HTTP fetching them, This way your application download size is smaller, and you only load the translations you need, when you need them. :wink: I use https://github.com/ngx-translate/http-loader to load my translations.

**Reply by SDemonUA** [reply]
For a front-end we are using `i18next-http-backend`, copying to assets just allows us to load them from statics server and not a backend. Tho for a server-side translations we would probably transit to something like `i18next-fs-backend` in the future.
